tests/ui/llm_studio_page.py

```diff
@@ -0,0 +1,242 @@
+        locator = self.page.get_by_role("button", name="View datasets")
+        locator.click()
+
+    def assert_dataset_deletion(self, dataset_name: str):
+        self.view_datasets()
+        dataset = self.page.get_by_role("button", name=dataset_name)
+        # Assert that the element not found
+        expect(dataset).not_to_be_visible()
+
+    def create_experiment(self, name: str):
+        # Get locator element
+        locator = self.page.get_by_role("button", name="Create experiment")
+        locator.click()
+        self.experiment_name(name)
+
+    def slider(self, slider_selector, target_value: str):
+        is_completed = False
+        i = 0
+        # Get the slider element
+        slider = self.get_by_test_id(slider_selector)
+        slider.click()
+        # Get the bounding box of the slider
+        bounding_box = slider.bounding_box()
+        x1 = bounding_box["x"]
+        y = bounding_box["y"] + bounding_box["height"] / 2
+
+        while not is_completed:
+            self.page.mouse.move(x1, y)
+            self.page.mouse.down()
+            x2 = bounding_box["x"] + bounding_box["width"] * float(i) / 100
+            self.page.mouse.move(x2, y)
+            self.page.mouse.up()
+            value_now = slider.get_attribute("aria-valuenow")
+
+            if value_now == target_value:
+                is_completed = True
+            else:
+                # Move the slider a little bit (adjust the step as needed)
+                step = 0.1  # Adjust this value based on your requirements
+                x1 = x2
+                i += step
+
+    def run_experiment(self):
+        self.page.wait_for_timeout(3000)
+        locator = self.get_by_test_id(self.EXPERIMENT_START_RUN_SELECTOR)
+        locator.click()
+        self.page.wait_for_timeout(3000)
+
+    def experiment_name(self, name: str):
+        locator = self.get_by_test_id(self.EXPERIMENT_NAME_SELECTOR)
+        locator.fill(name)
+
+    def llm_backbone(self, value: str):
+        locator = self.page.get_by_role("combobox", name="LLM Backbone")
+        locator.fill(value)
+
+    def data_sample(self, value):
+        self.slider(self.DATA_SAMPLING, value)
+
+    def max_length_prompt(self, value):
+        self.slider(self.MAX_LENGTH_PROMPT, value)
+
+    def max_length_answer(self, value):
+        self.slider(self.MAX_LENGTH_ANSWER, value)
+
+    def max_length(self, value):
+        self.slider(self.MAX_LENGTH, value)
+
+    def max_length_inference(self, value):
+        self.slider(self.MAX_LENGTH_INFERENCE, value)
+
+    def view_experiment_page(self):
+        locator = self.page.get_by_role("button", name="View experiments")
+        locator.click()
+
+    def view_experiment(self, experiment_name: str):
+        self.view_experiment_page()
+        i = self.find_experiment_index(experiment_name)
+        status = self.page.locator(
+            f"{self.EXPERIMENT_STATUS_SELECTOR} >> nth={i}"
+        ).inner_text()
+
+        while True:
```

Comment:
we can wait for locator with filters instead of re-implementing the poll loop
https://playwright.dev/python/docs/locators#filtering-locators